From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Thu, 17 Jul 2014 22:59:52 +0200
Subject: [204/249] staging: rtl8723au: Remove RegBcnCtrlVal
Origin: https://git.kernel.org/linus/e4d480b012f371c12a5c317494a5e068ad27216f

The value saved in RegBcnCtrlVal was never used. In addition the code
was highly endian broken, declaring it as a u32, then accessing it via
a u8 * later.

Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/staging/rtl8723au/hal/rtl8723a_hal_init.c | 18 +++++-------------
 drivers/staging/rtl8723au/include/rtl8723a_hal.h  |  1 -
 2 files changed, 5 insertions(+), 14 deletions(-)

Index: linux-3.16.7/drivers/staging/rtl8723au/hal/rtl8723a_hal_init.c
===================================================================
--- linux-3.16.7.orig/drivers/staging/rtl8723au/hal/rtl8723a_hal_init.c
+++ linux-3.16.7/drivers/staging/rtl8723au/hal/rtl8723a_hal_init.c
@@ -882,20 +882,13 @@ void rtl8723a_read_chip_version(struct r
 /*  */
 void SetBcnCtrlReg23a(struct rtw_adapter *padapter, u8 SetBits, u8 ClearBits)
 {
-	struct hal_data_8723a *pHalData;
-	u32 addr;
-	u8 *pRegBcnCtrlVal;
+	u8 val8;
 
-	pHalData = GET_HAL_DATA(padapter);
-	pRegBcnCtrlVal = (u8 *)&pHalData->RegBcnCtrlVal;
+	val8 = rtl8723au_read8(padapter, REG_BCN_CTRL);
+	val8 |= SetBits;
+	val8 &= ~ClearBits;
 
-	addr = REG_BCN_CTRL;
-
-	*pRegBcnCtrlVal = rtl8723au_read8(padapter, addr);
-	*pRegBcnCtrlVal |= SetBits;
-	*pRegBcnCtrlVal &= ~ClearBits;
-
-	rtl8723au_write8(padapter, addr, *pRegBcnCtrlVal);
+	rtl8723au_write8(padapter, REG_BCN_CTRL, val8);
 }
 
 void rtl8723a_InitBeaconParameters(struct rtw_adapter *padapter)
@@ -903,7 +896,6 @@ void rtl8723a_InitBeaconParameters(struc
 	struct hal_data_8723a *pHalData = GET_HAL_DATA(padapter);
 
 	rtl8723au_write16(padapter, REG_BCN_CTRL, 0x1010);
-	pHalData->RegBcnCtrlVal = 0x1010;
 
 	/*  TODO: Remove these magic number */
 	rtl8723au_write16(padapter, REG_TBTT_PROHIBIT, 0x6404);	/*  ms */
Index: linux-3.16.7/drivers/staging/rtl8723au/include/rtl8723a_hal.h
===================================================================
--- linux-3.16.7.orig/drivers/staging/rtl8723au/include/rtl8723a_hal.h
+++ linux-3.16.7/drivers/staging/rtl8723au/include/rtl8723a_hal.h
@@ -356,7 +356,6 @@ struct hal_data_8723a {
 	u8	fw_ractrl;
 	u8	RegTxPause;
 	/*  Beacon function related global variable. */
-	u32	RegBcnCtrlVal;
 	u8	RegFwHwTxQCtrl;
 	u8	RegReg542;
 
